ios/objc/apis/TorchIValue.h

```diff
@@ -0,0 +1,51 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, TorchIValueType) {
+    TorchIValueTypeNone,
+    TorchIValueTypeTensor,
+    TorchIValueTypeBool,
+    TorchIValueTypeDouble,
+    TorchIValueTypeInt,
+    TorchIValueTypeBoolList,
+    TorchIValueTypeIntList,
+    TorchIValueTypeDoubleList,
+    TorchIValueTypeTensorList,
+};
+
+@class TorchTensor;
+
+@interface TorchIValue : NSObject
```

Comment:
I think here we are trying to mirror the `at::IValue` class name in C++ frontend, and IMO it would be a good idea to have them consistent to avoid confusion.